src/main/java/org/apache/sysds/runtime/compress/colgroup/dictionary/DeltaDictionary.java

```diff
@@ -20,26 +20,90 @@
 package org.apache.sysds.runtime.compress.colgroup.dictionary;
 
 import org.apache.commons.lang.NotImplementedException;
-import org.apache.sysds.runtime.functionobjects.Divide;
-import org.apache.sysds.runtime.functionobjects.Minus;
-import org.apache.sysds.runtime.functionobjects.Multiply;
-import org.apache.sysds.runtime.functionobjects.Plus;
+import org.apache.sysds.runtime.compress.colgroup.indexes.IColIndex;
+import org.apache.sysds.runtime.data.SparseBlock;
+import org.apache.sysds.runtime.functionobjects.*;
+import org.apache.sysds.runtime.instructions.cp.CM_COV_Object;
+import org.apache.sysds.runtime.matrix.data.MatrixBlock;
+import org.apache.sysds.runtime.matrix.operators.BinaryOperator;
 import org.apache.sysds.runtime.matrix.operators.ScalarOperator;
+import org.apache.sysds.runtime.matrix.operators.UnaryOperator;
+
+import java.io.DataOutput;
+import java.io.IOException;
 
 /**
  * This dictionary class is a specialization for the DeltaDDCColgroup. Here the adjustments for operations for the delta
  * encoded values are implemented.
  */
-public class DeltaDictionary extends Dictionary {
+public class DeltaDictionary extends ADictionary {
+
 	private static final long serialVersionUID = -5700139221491143705L;
 	
 	private final int _numCols;
 
+	protected final double[] _values;
+
 	public DeltaDictionary(double[] values, int numCols) {
-		super(values);
+		_values = values;
 		_numCols = numCols;
 	}
 
+	@Override
+	public double[] getValues() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double getValue(int i) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double getValue(int r, int col, int nCol) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public long getInMemorySize() {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double aggregate(double init, Builtin fn) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double aggregateWithReference(double init, Builtin fn, double[] reference, boolean def) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double[] aggregateRows(Builtin fn, int nCol) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double[] aggregateRowsWithDefault(Builtin fn, double[] defaultTuple) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public double[] aggregateRowsWithReference(Builtin fn, double[] reference) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public void aggregateCols(double[] c, Builtin fn, IColIndex colIndexes) {
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public void aggregateColsWithReference(double[] c, Builtin fn, IColIndex colIndexes, double[] reference, boolean def) {
+		throw new NotImplementedException();
+	}
+
 	@Override
 	public DeltaDictionary applyScalarOp(ScalarOperator op) {
```

Comment:
This content of applyScalarOp is not completely valid, if it is not correct, comment it out and put a NotImplementedException.

Reply:
I will check it again before the next commit/push, the issue remains unresolved for now and not forgotten.